Examples/Tests/collision/analysis_collision_2d.py

```diff
@@ -41,14 +42,16 @@
 me = 9.10938356e-31
 mi = me * 5.0
 
+## In the first part of the test we verify that the output data is consistent with the exponential
+## fit.
+
 # exponential fit coefficients
 a =  0.041817463099883
 b = -0.083851393560288
 
 last_fn = sys.argv[1]
-temp = re.compile("([a-zA-Z_]+)([0-9]+)")
-res = temp.match(last_fn).groups()
-fn_list = glob(res[0] + "?????")
+if (last_fn[-1] == "/"): last_fn = last_fn[:-1]
+fn_list = glob(last_fn[:-5] + "?????")
```

Comment:
The regex function was not working for me locally, because I use a path instead of simply the name of the output folder in my command line argument (`python analysis_collision_2d.py diags/diag100150`). I've thus replaced it with these lines, which should hopefully be more robust when doing the test locally.

Reply:
Can we make a check to keep supporting the old one?
What exact call did you use? `python analysis_collision_2d.py diags/` ?

Reply:
No, I used the call given in my previous comment (`python analysis_collision_2d.py diags/diag100150`). Not adding `diag100150` results in a yt error `Supplied ('diags/',) {}, but could not load!`.

In any case, is there a reason to keep supporting the old one?
Currently (with the regex command) we expect the command line argument to be `diag100150` (or `collisionXYZ_plt00150` when run on Azure) with no path. When adding a path (e.g. when running locally), the regex command fails because of the character `/`.
The new solution should work both with and without a path, so we might as well keep it, no?

More generally, this raises the question of how much effort we want to put into making it easy to run the tests locally (cf. #1560). Personally I'm fine with the way things are. I like to directly run the tests and analysis script (when testing or developing stuff) and to do that I usually have to comment a few lines (e.g. `import checksumAPI`) and maybe sometimes set a boolean to true, which is ok for me. But in this specific case I think it's worth the effort because it's not too obvious how the regex command works which makes it non trivial to run the analysis script locally.